### DISCUSSION TITLE
Minimal status update

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16821] [General]
I am aware of the 'status' command but is there a minimalist way to get the status of outlets on a power plug. 

The limit the amount of communications between the broker and the device, I've set logging to 0 and every 5 minutes.
However, I'd like to know the status of only the outlets at all times for safety.
Is there a way to poll the device to get just the status of the outlet/s and not a bunch of other information?

**Comment by spcqike** [comment]
You can use 
http://ip/cm?cmnd=power

**Reply by sfromis** [reply]
Setting logging to 0 does not change the amount of traffic with the MQTT broker. Tasmota will by default report power changes as they happens, much lower traffic than the highly wasteful technique of polling.

**Reply by dac2020** [reply]
Yes, I see how much data keeps coming in even with logging disabled.
Mainly wondering if there is a way to limit that traffic to outlet status changes only.

http://ip/cm?cmnd=power
I'm using the mosquitto_sub and console. Not sure what this means since 'power' is not a command, is it?

**Reply by hipitihop** [reply]
@dac2020 It is a Tasmota command which can be invoked over http as @spcqike showed in his response.
Replace the`ip` in that URL with your plugs ip.

All commands available here: https://tasmota.github.io/docs/Commands/

**Reply by sfromis** [reply]
If you want to reduce the amount of MQTT traffic which is not related to specific events happening, you can set a long TelePeriod, or set it to 0 to disable the telemetry messages. Again, using polling is just a bad method, when you want it "at all times for safety". Really excessive amount of traffic if you mean that seriously. Via MQTT, Tasmota will publish at the moment of something changing.

**Reply by dac2020** [reply]
@hipitihop
Sure, it can be done using http but as I said, I'm not using http :). Only console and mosquitto client.
I am aware of the commands list and it shows the following;
```
Every command used without a parameter (payload) returns the current setting.
    Power returns the status of first defined power output (usually Relay1)
```
What I've not found is if a device could send only the status of its outlets/relays or be polled for that without additional data.
I can parse it but just wondered if there is something I'm not finding.

@sfromis
Do you mean just let the device communicate with the broker and get the info from that?
That's what I'm doing now but the problem is that the device only communicates with the broker when a pub or console command is issued. If I'm remote, how can I know instantly what the status of all the outlets are other than issuing another pub or console command? The only way would be using a DB or polling it no?

**Reply by sfromis** [reply]
Typical example: When I have a plug with a relay and a momentary push button, I can obviously switch on or off the relay using commands via MQTT. If I instead push the hardware button, the Tasmota web UI toggle, or use the Power command, in all 3 cases Tasmota will publish the new Power state as it happens. Much lower traffic than polling constantly.

**Reply by barbudor** [reply]
@dac2020 There is neever too much data in STATE and SENSOR message if you are serious about monitoring your tasmotas

![tasmotas](https://user-images.githubusercontent.com/2996042/195979459-2eada1d6-0c82-4d49-b069-22b29b0349ce.png)

Now if you really really want only POWER state, just as sfromis said, set Teleperiod to 0 and set PowerRetain to 1 so the last power state is maintained the the broker so just have to mosquitto_sub to get the power status without polling

**Reply by dac2020** [reply]
@sfromis
```
in all 3 cases Tasmota will publish the new Power state as it happens. Much lower traffic than polling constantly.
```
@barbudor
```
l publish the new Power state as it happens. Much lower traffic than polling constantly.
```
Nice graph. I'm not at the point of combining all of the statuses from everything I've got running. 
I did fire up an HA instance now running in the background, just not using it yet.

Yes, I understand that the devices communicate with the broker so I can always see the status changes. 
I believe I usually set teleperiod to 5 mins and sometimes 0.
 
However, I was curious to know if there is a way to just get the status of the plugs without a bunch of extra information by actually polling the device but it does not seem there is a way of doing that.

**Reply by sfromis** [reply]
Sure you **can** do polling, as already suggested, it is just a bad strategy when you already get events as it happens, also without ever using the telemetry messages every 5 minutes. Of course, you'd generate a **lot** of traffic if wanting an updated status all the time, and still slower than getting the data via MQTT.

I can have current status on by backend simultaneously with the relay clicking, only with any transmission latency added, should normally be very fast. Also when leaving out telemetry data by having `TelePeriod 0`.

**Reply by dac2020** [reply]
Did I miss something?
>Sure you can do polling, as already suggested, it

The suggestion was to use 'cmnd=power' but;
```
Every command used without a parameter (payload) returns the current setting.
Power returns the status of first defined power output (usually Relay1)
```
I don't see a way to poll all outlet statuses only without more information. I have plugs with 2,4,5 outlets and 4 relay boards so need to see the status of all outlets/relays.

**Reply by barbudor** [reply]
`power0`

```
17:44:52.160 CMD: power0
17:44:52.166 MQT: stat/skt-pcvalou/POWER1 = {"POWER1":"OFF"}
17:44:52.170 MQT: stat/skt-pcvalou/POWER1 = OFF (retained)
17:44:52.173 MQT: stat/skt-pcvalou/POWER2 = {"POWER2":"OFF"}
17:44:52.175 MQT: stat/skt-pcvalou/POWER2 = OFF (retained)
17:44:52.178 MQT: stat/skt-pcvalou/POWER3 = {"POWER3":"OFF"}
17:44:52.180 MQT: stat/skt-pcvalou/POWER3 = OFF (retained)
17:44:52.183 MQT: stat/skt-pcvalou/POWER4 = {"POWER4":"OFF"}
17:44:52.185 MQT: stat/skt-pcvalou/POWER4 = OFF (retained)
```

**Reply by sfromis** [reply]
The Command `Power0` even gives a compact one-line output when used via http. Still a bad strategy if you want to keep a status updated.

And obviously, you should just never open the http port of Tasmota devices directly to the big bad Internet.

**Reply by dac2020** [reply]
Nice, I did miss that then. power0, power1, etc show the status of all or individual without a lot of other info. 
Useful now and then and wondered how it could be done.
Too bad I can't get that more compact one using broker/console but that's ok, I can live with this.

Thank you everyone for the help.
